test/cql-pytest/cassandra_tests/validation/operations/insert_update_if_condition_statics_test.py

```diff
@@ -0,0 +1,359 @@
+                           "UPDATE %s SET v = 'newVal2' WHERE id = 1 AND k = 'k2' IF v = 'val3'; " +
+                           "APPLY BATCH"),
+                   [row(False,1,"k2","newVal"),row(False,1,"k2","newVal")] if is_scylla else [row(False, 1, "k2", "newVal")])
+
+# Test CASSANDRA-10532
+def testStaticColumnsCasDelete(cql, test_keyspace, is_scylla):
+    with create_table(cql, test_keyspace, "(pk int, ck int, static_col int static, value int, PRIMARY KEY (pk, ck))") as table:
+        execute(cql, table, "INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 1, 2)
+        execute(cql, table, "INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 3, 4)
+        execute(cql, table, "INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 5, 6)
+        execute(cql, table, "INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 7, 8)
+        execute(cql, table, "INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 2, 1, 2)
+        execute(cql, table, "INSERT INTO %s (pk, static_col) VALUES (?, ?)", 1, 1)
+
+        assertRows(execute(cql, table, "DELETE static_col FROM %s WHERE pk = ? IF static_col = ?", 1, 2), row(False, 1))
+        assertRows(execute(cql, table, "DELETE static_col FROM %s WHERE pk = ? IF static_col = ?", 1, 1), row(True,1) if is_scylla else row(True))
+
+        assertRows(execute(cql, table, "SELECT pk, ck, static_col, value FROM %s WHERE pk = 1"),
+                   row(1, 1, null, 2),
+                   row(1, 3, null, 4),
+                   row(1, 5, null, 6),
+                   row(1, 7, null, 8))
+        execute(cql, table, "INSERT INTO %s (pk, static_col) VALUES (?, ?)", 1, 1)
+
+        # Scylla's and Cassandra's error messages here focus on different things
+        assertInvalid(cql, table,
+                             "DELETE static_col FROM %s WHERE ck = ? IF static_col = ?", 1, 1)
+
+        assertInvalidMessage(cql, table, "Invalid restrictions on clustering columns since the DELETE statement modifies only static columns",
+                             "DELETE static_col FROM %s WHERE pk = ? AND ck = ? IF static_col = ?", 1, 1, 1)
+
+        # Scylla's and Cassandra's error messages here are different,
+        # sharing only the word "primary key"
+        assert_invalid_message_ignore_case(cql, table, "primary key",
+                             "DELETE static_col, value FROM %s WHERE pk = ? IF static_col = ?", 1, 1)
+
+        # Same query but with an invalid condition
+        assert_invalid_message_ignore_case(cql, table, "primary key",
+                             "DELETE static_col, value FROM %s WHERE pk = ? IF static_col = ?", 1, 2)
+
+        # DELETE of an underspecified PRIMARY KEY should not succeed if static is not only restriction
+        assert_invalid_message_ignore_case(cql, table, "primary key",
+                             "DELETE static_col FROM %s WHERE pk = ? IF value = ? AND static_col = ?", 1, 2, 1)
+
+        print("NYH")
```

Comment:
huh?

Reply:
Probably a leftover debug msg. @nyh please send a follow-up to remove it. I won't dequeue this PR just for this.